src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs

```diff
@@ -7652,5 +7652,367 @@ public void RecordDeclaration()
+                "T? P",
+                SymbolDisplayPartKind.TypeParameterName,
+                SymbolDisplayPartKind.Punctuation,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+
+            format = format.WithMiscellaneousOptions(SymbolDisplayMiscellaneousOptions.None); // without SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier
+            Verify(
+                closedGenericMemmber.ToDisplayParts(format),
+                "Object P",
+                SymbolDisplayPartKind.ClassName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+            Verify(
+                openGenericMemmber.ToDisplayParts(format),
+                "T P",
+                SymbolDisplayPartKind.TypeParameterName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+        }
+
+        [Fact]
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        public void NullableAttributes_MayBeNullOnStructProperty()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+public class A<T>
+{
+    [MaybeNull]
+    public T P { get; }
+}
+
+public class Test: A<int> { }
+";
+            var compilation = CreateCompilation(new[] { source, MaybeNullAttributeDefinition });
+            var closedGenericMember = compilation.GetTypeByMetadataName("Test").BaseType().GetMember("P");
+            var format = new SymbolDisplayFormat(
+                memberOptions: SymbolDisplayMemberOptions.IncludeParameters | SymbolDisplayMemberOptions.IncludeType | SymbolDisplayMemberOptions.IncludeModifiers,
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                miscellaneousOptions: SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier);
+
+            Verify(
+                closedGenericMember.ToDisplayParts(format),
+                "Int32 P",
+                SymbolDisplayPartKind.StructName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+        }
+
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        [Fact]
+        public void NullableAttributes_VisualBasicReferencesCSharp()
+        {
+            var sourceCS = @"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+namespace N
+{
+    public class A<T>
+    {
+        [MaybeNull]
+        public T P { get; }
+    }
+}
+";
+            var sourceVB = @"
+Imports N
+Public Class Test
+    Inherits A(Of Object)
+End Class";
+            var compCS = CreateCompilation(new[] { sourceCS, MaybeNullAttributeDefinition });
+            var refCS = compCS.EmitToImageReference();
+            var compVB = CreateVisualBasicCompilation(GetUniqueName(), sourceVB, referencedAssemblies: new[] { MscorlibRef, refCS });
```

Comment:
>MscorlibRef [](start = 111, length = 11)

Why do we need to pass this reference explicitly? This is unexpected and is likely to cause issues with the test in the future. If there is some kind of reference mismatch between the compilations, I think we should ensure they use the same target framework instead.

Reply:
I replaced it with `TargetFrameworkUtil.GetReferences(TargetFramework.Standard, new[] { refCS })` in b6e79a8b. If I just drop MscorlibRef, `object` can't be resolved.